README.md

```diff
@@ -18,38 +18,172 @@ sources).
+
+    ```shell
+    cd build/tmp/deploy/images/genericx86-64/
+    sudo umount /dev/sdx*
+    sudo bmaptool copy tb-minimal-image-genericx86-64.rootfs.wic.gz /dev/sdx
+    ```
+
+    and you should see output similar to this (the exact size number may differ):
+
+    ```shell
+    bmaptool: info: block map format version 2.0
+    bmaptool: info: 275200 blocks of size 4096 (1.0 GiB), mapped 73240 blocks (286.1 MiB or 26.6%)
+    bmaptool: info: copying image 'tb-minimal-image-genericx86-64.rootfs.wic.gz' to block device '/dev/sdx' using bmap file 'tb-minimal-image-genericx86-64.rootfs.wic.bmap'
+    bmaptool: info: 100% copied
+    bmaptool: info: synchronizing '/dev/sdx'
+    bmaptool: info: copying time: 19.3s, copying speed 14.9 MiB/sec
+    ```
+
+## Booting
+
+To run TrenchBoot connect drive with flashed image to target platform and boot
+from it. In GRUB menu you can choose normal `boot` or `skl-boot`.
+
+```text
+                             GNU GRUB  version 2.06
+
+ +----------------------------------------------------------------------------+
+ |*boot                                                                       |
+ | skl-boot                                                                   |
+ |                                                                            |
+ |                                                                            |
+ |                                                                            |
+ +----------------------------------------------------------------------------+
+
+      Use the ^ and v keys to select which entry is highlighted.
+      Press enter to boot the selected OS, `e' to edit the commands
 ```
 
-Then flash the drive with bmap-tools:
+After a while you should see login prompt.
 
-```bash
-bmaptool copy --bmap tb-minimal-image-pcengines-apu2.wic.bmap tb-minimal-image-pcengines-apu2.wic.gz /dev/<dev>
+```text
+early console in extract_kernel
+input_data: 0x0000000006801548
+input_len: 0x000000000121e953
+output: 0x0000000004600000
+output_len: 0x00000000033caee8
+kernel_total_size: 0x0000000003030000
+needed_size: 0x0000000003400000
+trampoline_32bit: 0x0000000000000000
+Physical KASLR using RDRAND RDTSC...
+Virtual KASLR using RDRAND RDTSC...
+
+Decompressing Linux... Parsing ELF... Performing relocations... done.
+Booting the kernel (entry_offset: 0x0000000000000000).
+
+
+Reference Yocto distro for PC Engines hardware 0.2.0 tb ttyS0
+
+tb login:
 ```
 
-Example output
+To login use `root` account without password.
+
+## Running in QEMU
 
+It's possible to test image by running it in QEMU. Depending on QEMU
+configuration not all features may be available, SKL boot among others.
+
+To create bootable image you can once again use `bmaptool` but this time instead
+of writing image to flash device you write it to file
+
+```shell
+cd build/tmp/deploy/images/genericx86-64
+bmaptool copy tb-minimal-image-genericx86-64.rootfs.wic.gz trenchboot.img
```

Comment:
@m-iwanicki Whe do we use bmaptool here? Is it considerably faster than simply unpacking this archive?

Reply:
Don't we generate the `tb-minimal-image-genericx86-64.rootfs.wic` as well? In such a case, no need to to either of that, simply use the .wic image in QEMU.

Reply:
You are right, we do and we can use it I must've missed it. Fixed in this PR: https://github.com/3mdeb/meta-trenchboot/pull/30